src/modules/persistent-store.ts

```diff
@@ -33,10 +31,21 @@ export class PersistentStore<T> extends Store<T> {
     return super.getState();
   }
 
+  setState(...args: Parameters<Store<T>['setState']>) {
+    if (!this.isReady) {
+      if (process.env.NODE_ENV === 'development') {
+        throw new Error(
+          'You are trying to write to a PersistentStore before {ready}'
+        );
+      }
+    }
+    return super.setState(...args);
```

Comment:
Are we ok with writing if this.isReady is false? Cause in production it looks like we are going to write anyway